librz/bin/bobj.c

```diff
@@ -817,48 +783,60 @@ RZ_API const RzList *rz_bin_object_get_resources(RzBinObject *obj) {
 /**
  * \brief Remove all previously identified strings in the binary object and scan it again for strings.
  */
-RZ_API const RzList *rz_bin_object_reset_strings(RzBin *bin, RzBinFile *bf, RzBinObject *obj) {
-	rz_return_val_if_fail(bin && bf && obj, NULL);
-	if (obj->strings) {
-		rz_list_free(obj->strings);
-		obj->strings = NULL;
-	}
+RZ_API bool rz_bin_object_reset_strings(RzBin *bin, RzBinFile *bf, RzBinObject *obj) {
+	rz_return_val_if_fail(bin && bf && obj, false);
+	RZ_FREE_CUSTOM(obj->strings, rz_bin_string_database_free);
 
+	RzList *strings = NULL;
 	RzBinPlugin *plugin = obj->plugin;
 	if (plugin && plugin->strings) {
-		obj->strings = plugin->strings(bf);
+		strings = plugin->strings(bf);
 	} else {
 		// when a bin plugin does not provide it's own strings
 		// we always take all the strings found in the binary
 		// the method also converts the paddrs to vaddrs
-		obj->strings = rz_bin_file_strings(bf, bin->minstrlen, true);
+		strings = rz_bin_file_strings(bf, bin->minstrlen, true);
 	}
-	if (bin->debase64) {
-		rz_bin_object_filter_strings(obj);
+
+	// RzBinStrDb becomes the owner of the RzList strings
+	obj->strings = rz_bin_string_database_new(strings, obj->opts.loadaddr, bin->debase64);
+	return obj->strings != NULL;
+}
+
+/**
+ * \brief Return RzBinString if at \p address \p there is an entry in the RzBinObject string database
+ */
+RZ_API RzBinString *rz_bin_object_string_at(RZ_NONNULL RzBinObject *obj, ut64 address, bool is_va) {
+	rz_return_val_if_fail(obj, false);
+	if (is_va) {
+		return ht_up_find(obj->strings->virt, address, NULL);
 	}
-	return obj->strings;
+	return ht_up_find(obj->strings->phys, address, NULL);
```

Comment:
`obj->strings` can be NULL from what I can tell. Same goes for all other places that access it and don't check.